File: lab2/Shop/Shop/Category.cs
using System;
using System.Collections.Generic;

namespace Shop
{
    public class Category
    {
        public Category() {
            Products = new List<Product>();
        }

        public int CategoryId { get; set; }
        public string Name { get; set; }
        public List<Product> Products { get; set; }
    }
}

File: lab3/src/main/java/lab3/model/Category.java
package lab3.model;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity(name="categories")
public class Category {
    @Id
    private String name;

    @OneToMany(mappedBy = "category")
    private List<Product> products;

    public Category() {}

    public Category(String name) {
        this.name = name;
        this.products = new ArrayList<>();
    }

    public String getName() {
        return name;
    }

    public void addProduct(Product product) {
        products.add(product);
    }

    public List<Product> getProducts() {
        return products;
    }
}